compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3877,7 +3874,8 @@ private ParserRuleContext getNextRuleForKeywords(ParserRuleContext currentCtx, i
                     return ParserRuleContext.CONFLICT_KEYWORD;
                 } else if (parentCtx == ParserRuleContext.ON_CLAUSE) {
                     return ParserRuleContext.EXPRESSION;
-                } else if (parentCtx == ParserRuleContext.ON_FAIL_CLAUSE) {
+                } else if (parentCtx == ParserRuleContext.ON_FAIL_CLAUSE ||
+                            parentCtx == ParserRuleContext.ON_FAIL_CHECK) {
```

Comment:
We don't start a context for `ON_FAIL_CHECK`. We do for `CHECK_EXPRESSION`
```suggestion
                            parentCtx == ParserRuleContext.CHECK_EXPRESSION) {
```